src/Microsoft.ML.Data/EntryPoints/Cache.cs

```diff
@@ -34,7 +34,7 @@ public sealed class CacheOutput
         }
 
         [TlcModule.EntryPoint(Name = "Transforms.DataCache", Desc = "Caches using the specified cache option.", UserName = "Cache Data")]
-        public static CacheOutput CacheData(IHostEnvironment env, CacheInput input)
+        public static CacheOutput CacheData(IHostEnvironment env, CacheInput input, IFileHandle fileHandle)
```

Comment:
Since this is an `EntryPoint` itself, I don't know if we can add parameters to it. I'm not 100% exactly familiar with how EntryPoints work.

@ganik ?

Reply:
Maybe it could work something like this:

The `CacheInput` can take a path to where on the file system to put the cached file. And then if disk caching was specified the `CacheOutput` will have the path to the file. It is then up to the caller to clean the file up when they are done with it.

Thoughts?

---
In reply to: [256896499](https://github.com/dotnet/machinelearning/pull/2539#discussion_r256896499) [](ancestors = 256896499)

Reply:
comment for next PR